### DISCUSSION TITLE
How to Select All (or more than one screenfull) of a terminal window?

### DISCUSSION THREAD

**Discussion by LorenAmelang** [holzschu/a-shell#815] [Q&A]
I often have long command results or directories that I'd like to save to an external text file. If the text is all visible at once, I can tap a word and then drag the little selection handles to choose what I want. But unlike some iOS apps, A-Shell won't let me drag them beyond the current screen - it won't scroll the offscreen text into view. And there is never a Select All option in the actions bar. 

Am I missing some trick? Actually, Select All would be better than drag select, it is easy to delete unwanted text in a text editor.

**Comment by holzschu** [comment]
It's a good question; that is a balance between ability to select and ability to scroll. In the previous version, iOS was constantly interpreting finger movements as scrolling, which made selecting a single word difficult. Now I've disabled scrolling while selecting, which makes it possible to select things on the current screen. The drawback, as you say, is that it's impossible to select text that is beyond the current screen. I'll need to revisit the selection system, but in the meantime I recommend simply redirecting the output of your command directly to a file:
```sh
myCommand > myFile
```
will just write the output of `myCommand` in a file. If you want to also see the results on screen, there's the command `tee`:
```sh
myCommand | tee myFile
```
will both print the result of `myCommand` on the screen, and save it inside `myFile` as well.

**Reply by LorenAmelang** [reply]
Thanks! That actually works, saves myFile in ~/Documents . 

I found a note from ages ago: 
---
“Long press Win+V menu” popup:
(all commands look like Command+Key, but maybe Win+Key for me?)
K clear screen
N new window
W close window
F find
G find next
Shift+G find previous
X cut
V paste
A select all
+ bigger text
- smaller text
---

Not sure where I learned that...  But it requires an external keyboard on my iPad with a Win key (or I suppose an Apple Command key). And now it requires a long press of my Win key where nothing happens, followed by another long press of the Win key that pops up the menu…  

Or I can just use Win+A and immediately the whole terminal is selected! But none of the onscreen keyboards currently on my iPad have any Win or Command key. Is there some trick to get that menu without an external keyboard? 

Actually, what I might like best of all would be an option like my Linux Tabby Terminal - all my terminal activity is automatically logged to a dated log file that I can refer back to whenever I want to remember something precisely. Could you add that feature?

**Comment by marcioflores01** [comment]
I finally figured it out after a long discussion with Perplexity, it finally admitted: you are  absolutely correct—the lower handle in a-Shell’s terminal does not extend selection downward  (or at all, per your test), forcing reliance on the upper handle dragged down.[github]
Precise Selection Mechanics
•	Long-press word → handles appear (upper above start, lower at end).
•	Upper handle only works: Drag it up (selects prior lines) or down (selects following lines).
•	Lower handle: Effectively inert for extension—drags may reposition the end pf that line minimally but won’t reliably add lines below.
Not a Bug, it says  But suggests I Report Anyway
This stems from the dev’s scroll-disable fix, creating asymmetric handles as a side effect. It told me to 
File a GitHub issue at github.com/holzschu/a-shell/issues describing the inert lower handle + your iOS/a-Shell versions for potential tweak.[github +1]. 

So there you have it!
